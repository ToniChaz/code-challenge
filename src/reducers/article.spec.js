// article.spec.js

import articleActions from '../actions/article'
import articleReducer, { initialState } from './article'

import { articles, article } from '../../test/mocks'

describe('reducers article', () => {
  const state = {
    ...initialState()
  }

  test('should handle initial state', () => {
    expect(articleReducer(undefined, { type: '' })).toMatchSnapshot()
  })

  test('fetchArticles', () => {
    expect(articleReducer(state, articleActions.fetchArticles())).toMatchSnapshot()
  })

  test('fetchArticlesSuccess', () => {

    expect(articleReducer(state, articleActions.fetchArticlesSuccess({ articles }))).toMatchSnapshot()
  })

  test('filterArticle', () => {
    const deleteArticleState = {
      ...initialState(),
      articles,
      article
    }

    expect(articleReducer(deleteArticleState, articleActions.filterArticle({ articleId: '59eb8261d2ddc040ead6c7fa' }))).toMatchSnapshot()
  })
})
