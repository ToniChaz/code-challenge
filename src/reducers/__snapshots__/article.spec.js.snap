exports[`reducers article fetchArticles 1`] = `
Object {
  "article": null,
  "articles": null,
  "error": null,
  "loading": true,
}
`;

exports[`reducers article fetchArticlesSuccess 1`] = `
Object {
  "article": null,
  "articles": Object {
    "articles": Array [
      Object {
        "_id": "59eb8261d2ddc040ead6c7fa",
        "author": "Bertram Wintheiser",
        "content": "Et iure enim saepe. Repellat reprehenderit sequi id. Eveniet sit ex. Rem et labore temporibus dicta architecto reprehenderit qui. Consequatur amet aspernatur porro.",
        "excerpt": "Et iure enim saepe. Repellat reprehenderit sequi id. Eveniet sit ex. Rem et labore temporibus dicta ",
        "published": false,
        "tags": Array [
          "Avon portal Outdoors",
          "networks Berkshire Fish",
        ],
        "title": "Senior Communications Coordinator",
      },
      Object {
        "_id": "59eb8261d2ddc040ead6c7fb",
        "author": "Asha Koelpin DDS",
        "content": "Hic ut quia ipsam. Atque neque nam molestiae earum explicabo ab. Molestias soluta molestiae rem similique tenetur. Maiores doloribus rerum corrupti sed voluptatum nostrum delectus facere et.",
        "excerpt": "Hic ut quia ipsam. Atque neque nam molestiae earum explicabo ab. Molestias soluta molestiae rem simi",
        "published": true,
        "tags": Array [
          "New Mexico moderator Practical",
          "Ethiopian Birr",
        ],
        "title": "Internal Tactics Supervisor",
      },
    ],
  },
  "error": null,
  "loading": false,
}
`;

exports[`reducers article filterArticle 1`] = `
Object {
  "article": Array [],
  "articles": Array [
    Object {
      "_id": "59eb8261d2ddc040ead6c7fa",
      "author": "Bertram Wintheiser",
      "content": "Et iure enim saepe. Repellat reprehenderit sequi id. Eveniet sit ex. Rem et labore temporibus dicta architecto reprehenderit qui. Consequatur amet aspernatur porro.",
      "excerpt": "Et iure enim saepe. Repellat reprehenderit sequi id. Eveniet sit ex. Rem et labore temporibus dicta ",
      "published": false,
      "tags": Array [
        "Avon portal Outdoors",
        "networks Berkshire Fish",
      ],
      "title": "Senior Communications Coordinator",
    },
    Object {
      "_id": "59eb8261d2ddc040ead6c7fb",
      "author": "Asha Koelpin DDS",
      "content": "Hic ut quia ipsam. Atque neque nam molestiae earum explicabo ab. Molestias soluta molestiae rem similique tenetur. Maiores doloribus rerum corrupti sed voluptatum nostrum delectus facere et.",
      "excerpt": "Hic ut quia ipsam. Atque neque nam molestiae earum explicabo ab. Molestias soluta molestiae rem simi",
      "published": true,
      "tags": Array [
        "New Mexico moderator Practical",
        "Ethiopian Birr",
      ],
      "title": "Internal Tactics Supervisor",
    },
  ],
  "error": null,
  "loading": true,
}
`;

exports[`reducers article should handle initial state 1`] = `
Object {
  "article": null,
  "articles": null,
  "error": null,
  "loading": true,
}
`;
